### PR TITLE
feat: [#449] Implement HTTP response stream 

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

RelatedTo https://github.com/goravel/goravel/issues/449

@coderabbitai summary

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: contracts/http/response.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type ContextResponse interface {
 	// Status sets the HTTP response status code and returns the ResponseStatus.
 	Status(code int) ResponseStatus
 	// Stream sends a streaming response with the specified status code and the given reader.
-	Stream(step func(w StreamWriter) error, code int, contentType string) Response
+	Stream(step func(w StreamWriter) error, code int) Response
```
Comment (hwbrzzl) [inline contracts/http/response.go:37]: Keep consistent with the `Json` method, `code` is the first paramter.

```suggestion
	Stream(code int, step func(w StreamWriter) error) Response
```

And add the `Stream` to `ResponseStatus`:

```
Stream(step func(w StreamWriter) error) Response
```

**Review by hwbrzzl — COMMENTED** [review]
I ran this feature locally, the strings do not be printed one by one, did I miss something?

```
facades.Route().Get("/stream", func(ctx http.Context) http.Response {
  b := []string{"a", "b", "c"}
  return ctx.Response().Stream(func(w http.StreamWriter) error {
    for _, a := range b {
      if _, err := w.Write([]byte(a)); err != nil {
	return err
      }
      
      if err := w.Flush(); err != nil {
	return err
      }
      
      time.Sleep(1 * time.Second)
    }
    
    return nil
  }, 200)
})
```

**Comment by kkumar-gcc** [comment]
> I ran this feature locally, the strings do not be printed one by one, did I miss something?
> 
> ```
> facades.Route().Get("/stream", func(ctx http.Context) http.Response {
>   b := []string{"a", "b", "c"}
>   return ctx.Response().Stream(func(w http.StreamWriter) error {
>     for _, a := range b {
>       if _, err := w.Write([]byte(a)); err != nil {
> 	return err
>       }
>       
>       if err := w.Flush(); err != nil {
> 	return err
>       }
>       
>       time.Sleep(1 * time.Second)
>     }
>     
>     return nil
>   }, 200)
> })
> ```


Buffered content only be displayed after reaching to **newline**.


```go
...
if _, err := w.Write([]byte(a + "\n")); err != nil {
     return err
}
...
```

Ref: https://stackoverflow.com/a/1716621/17666299

File: contracts/http/response.go (hunk 1 above)
Comment (kkumar-gcc) [reply contracts/http/response.go:37]: Ok

**Comment by hwbrzzl** [comment]
@kkumar-gcc I changed it, but doesn't have an effect.

```
facades.Route().Get("/stream", func(ctx http.Context) http.Response {
		b := []string{"a", "b", "c", "c", "c", "c", "c"}
		return ctx.Response().Stream(func(w http.StreamWriter) error {
			for _, a := range b {
				if _, err := w.Write([]byte(a + "\n")); err != nil {
					return err
				}

				if err := w.Flush(); err != nil {
					return err
				}

				time.Sleep(1 * time.Second)
			}

			return nil
		}, 200)
	})
```

**Review by hwbrzzl — APPROVED** [review]
Great

**Pull request merged**